### ISSUE TITLE
Target Size and standard `<label>` element behaviour - does it count?

### ISSUE THREAD

**Issue by patrickhlauke** [w3c/wcag#3440]
In most (all?) browsers, a correctly associated `<label>` for a form control acts as an additional clickable/tappable target area (though this behaviour is not explicitly specified in https://html.spec.whatwg.org/multipage/forms.html#the-label-element - it falls under platform convention).

For the purposes of calculating the target size, should auditors take the `<label>` into consideration? I'd argue that yes, they should. If we're agreed that it's yes, this should be mentioned in the understanding document for both 2.5.8 Target Size (Minimum) and 2.5.5 Target Size (Enhanced).

Thoughts?

**Comment by stevefaulkner** [comment]
@patrickhlauke i had assumed they would be and factored it in to my [target size bookmarklet ](https://html5accessibility.com/stuff/2023/08/28/quick-and-very-dirty-target-size-checker/)

**Comment by patrickhlauke** [comment]
yeah, likewise controls/interactions where JS is used to expand the actual clickable area (by making other elements clickable too)

**Comment by alastc** [comment]
Yep, that came up in the discussions because it could be a separate area. The definition of target ("region of the display") allows for that, so it is included.

**Comment by patrickhlauke** [comment]
perfect, i suspected as much. I may think about some addition/note for the understanding that makes this clear, just in case

**Comment by mbgower** [comment]
Just to revisit this discussion.
I think the issue that needs to be determined is if we treat a checkbox and its label as two separate inputs, or if together they form a single (potentially non-contiguous) target. I have been interpreting this as the latter -- the input and its label form one target, whether contiguous or not.

The rationale for this is as follows:
- clicking either the input or its label produces the identical result on the same input (i.e., the checked state will toggle identically regardless of where clicked)
- the label is conceptually part of the checkbox UIC ("a part of the content that is perceived by users as a single control for a distinct function")
- our definition of target specifies a UIC ("region of the display that will accept a pointer action, such as the interactive area of a user interface component")
- we discussed scenarios where a single target can be non-contiguous (e.g., you have a button that is 44x44 pixels, and you put a button that is 24x44 pixels across the centre of it; there are still only two targets, although there are now 3 distinct hit areas on the screen)

@stevefaulkner I note that your target size bookmarklet treats them as two separate targets. In the below screenshot of my straight-html form, each of the inputs and its label are shown as separate targets, and there's even a case where the proximity of a checkbox to its short label seems to cause an overlap failure between them (car label and car checkbox)
<img width="1180" alt="Screenshot 2023-11-15 at 8 59 48 AM" src="https://github.com/w3c/wcag/assets/14298245/a8e8142d-8202-434c-b38b-9d0e47052126">

I'm hoping we can agree that we wouldn't want a set of checkboxes labelled "a", "b", "c" to fail based on the center of the 1-character label being less than 12 pixels from the edge of the checkbox!

I guess we do have the wiggle room of saying that it can be met through either the "equivalent" or "user agent control" exceptions, but does anyone feel that's necessary?

**Comment by fstrr** [comment]
I stumbled across this thread in trying to decide if an "information" `button` placed next to a `label` could fail this criterion. There's no `label` related content in [Target Size (Min)](https://www.w3.org/WAI/WCAG22/Understanding/target-size-minimum.html) or [Enhanced](https://www.w3.org/WAI/WCAG22/Understanding/target-size-enhanced.html).

Here's a quick [codepen demo of the label / button](https://codepen.io/fstrr/pen/JoojLMe) I was looking at. I _think_ this fails due to the proximity of the two elements and the size of the `button`.

I can take a go at adding some content to the Understanding document once I get my other backlog taskforce items out of the way.

**Comment by mbgower** [comment]
The information button is a separate target from the input. It should have at least a 24x24 pixel target, or sufficient space from the neighbouring target.

I concur that I would fail this. HOWEVER, it is an interesting discussion point, in that the author has not set the target on the label, right? There is no need for that label to be part of the target; the input is sufficiently large on its own. So it's an unfortunate situation for the author.

**Comment by fstrr** [comment]
Okay cool. Thanks. I might make a PR and add this to the Understanding document as it's a fairly common design pattern.

**Comment by fstrr** [comment]
> I guess we do have the wiggle room of saying that the phenomenon of the input and its label being two separate targets can be met through either the "equivalent" or "user agent control" exceptions, but does anyone feel that's necessary?

It's worth having in the Understanding document, I think. Radio buttons are probably more more-common example than checkboxes for A, B, C options.
